synfig-studio/src/gui/widgets/widget_timeslider.cpp

```diff
@@ -384,29 +384,45 @@ Widget_Timeslider::on_motion_notify_event(GdkEventMotion* event) //for dragging
 }
 
 bool
-Widget_Timeslider::on_scroll_event(GdkEventScroll* event) //for zooming
+Widget_Timeslider::on_scroll_event(GdkEventScroll* event) //for zooming/moving timeline-bar
 {
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
 	etl::handle<TimeModel> &time_model = time_plot_data->time_model;
 
 	if (!time_model || get_width() <= 0 || get_height() <= 0)
 		return false;
 
-	Time time = time_plot_data->get_t_from_pixel_coord(event->x);
-
-	switch(event->direction) {
-	case GDK_SCROLL_UP: //zoom in
-		time_model->zoom(zoominfactor, time);
-		return true;
-	case GDK_SCROLL_DOWN: //zoom out
-		time_model->zoom(zoomoutfactor, time);
-		return true;
+	Time scroll_time = time_model->get_time(); //scroll is based on track time
+        Time zoom_time = time_plot_data->get_t_from_pixel_coord(event->x); //zoom is based on time represented by pixel 
```

Comment:
Please note here, after some thought I kept the separate time representation for scroll and zoom functionalities. 
As zooming would ideally remain unchanged when based on  `time_plot_data->get_t_from_pixel_coord` but for scrolling I felt it better suited with `time_model->get_time()` as calculation based on actual timeline-bar position on panel, kindly let me know your thoughts.